p2p/discovery/mdns.go

```diff
@@ -111,34 +75,38 @@ func NewMdnsService(ctx context.Context, peerhost host.Host, interval time.Durat
 }
 
 func (m *mdnsService) Close() error {
-	return m.server.Shutdown()
+	m.server.Shutdown()
+	// grandcat/zerconf swallows error, satisfy interface
+	return nil
 }
 
 func (m *mdnsService) pollForEntries(ctx context.Context) {
 
 	ticker := time.NewTicker(m.interval)
 	for {
+		resolver, err := mdns.NewResolver(nil)
```

Comment:
I think it might be beast to put resolver in `mdnsService` struct and reuse it, instead of creating a new one every iteration.

Reply:
Yikes, good catch. Fixed.